client/app/services/project.service.js

```diff
@@ -0,0 +1,108 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('taskingManager')
+        .service('projectService', [projectService]);
+
+    /**
+     * @fileoverview This file provides a project service.
+     * It generates the task grid for the project using Turf.js (spatial analysis)
+     * The task grid matches up with OSM's grid.
+     * Code is similar to Tasking Manager 2 (where this was written server side in Python)
+     */
+    function projectService() {
+
+        // Maximum resolution of OSM
+        var MAXRESOLUTION = 156543.0339;
+
+        // X/Y axis limit
+        var MAX = MAXRESOLUTION * 256 / 2;
+
+        // Target projection for Turf.js
+        var TARGETPROJECTION = 'EPSG:4326';
+
+        // Map projection in OpenLayers
+        var MAPPROJECTION = 'EPSG: 3857';
+
+        var service = {
+            getTaskGrid: getTaskGrid
+        };
+
+        return service;
+
+        /**
+         * Creates a task grid with features for a polygon feature.
+         * It snaps to the OSM grid
+         * @param areaOfInterest (ol.Feature) - this should be a polygon
+         * @param zoom
+         * @returns {Array} of Features
+         */
+        function getTaskGrid(areaOfInterest, zoom) {
```

Comment:
As discussed, we really should have some test here to ensure no-one breaks this algo, so that would be v. BAD